Fix 0.20.2 bundle versions, update SHAs

dfarrell07 opened the pull request:
See commit messages for details.

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

dfarrell07 committed:
Update bundle for pre-0.20.2

Updates operator bundle to version 0.20.2 following the same pattern
used successfully in release-0.21.

Bundle format matches 0.21:
- CSV content uses v prefix: "version": "v0.20.2"
- Bundle name: submariner.v0.20.2
- Konflux Dockerfile labels updated accordingly

Generated using VERSION=v0.20.2 with local git tag v0.20.2.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Update bundle SHAs to latest

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 commented:
I think we should make the bundle validation test not-required while we're figuring out the workflow.

skitt commented:
> It turns out we need to create a temporary local tag for all the versions to get set correctly when generating the bundle. See commit messages for details.

`make bundle` expects a version without a leading `v`: `make bundle VERSION=0.20.2`. Then you don’t need a tag.

dfarrell07 committed:
Use plain semver in CSV version field per feedback

Changed approach from using 'v0.20.2' to '0.20.2' in CSV version fields
to match standard semantic versioning format and be consistent with
csv-version Dockerfile label.

This diverges from 0.21, which had 'v0.21.0' in CSV content but '0.21.0'
in csv-version label.

Generated by:

    make bundle LOCAL_BUILD=1 VERSION=0.20.2

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 commented:
> > It turns out we need to create a temporary local tag for all the versions to get set correctly when generating the bundle. See commit messages for details.
> 
> `make bundle` expects a version without a leading `v`: `make bundle VERSION=0.20.2`. Then you don’t need a tag.

My goal was to match 0.21's result, but we can try this and hope OPM accepts it when rendering the FBC.

skitt commented:
I’m curious about the explanation in the last commit — what was the `Makefile` bug?

dfarrell07 commented:
> I’m curious about the explanation in the last commit — what was the `Makefile` bug?

I meant this:

```
  -	sed -i -e 's/$$(VERSION)/$(VERSION)/g' bundle/manifests/submariner.clusterserviceversion.yaml
  +	sed -i -e 's/$$(VERSION)/$(BUNDLE_VERSION)/g' bundle/manifests/submariner.clusterserviceversion.yaml
```

But we don't need that to get past is-semantic-version when we don't use a "v" prefix. I removed it from the commit message.
